lib/plugins/aws/invokeLocal/index.js

```diff
@@ -443,113 +422,107 @@ class AwsInvokeLocal {
       'invokeLocal',
       'artifact'
     );
-    return decompress(artifact, destination).then(() => destination);
+    await decompress(artifact, destination);
+    return destination;
   }
 
   getEnvVarsFromOptions() {
-    const envVarsFromOptions = {};
     // Get the env vars from command line options in the form of --env KEY=value
-    [].concat(this.options.env || []).forEach(itm => {
+    return [].concat(this.options.env || []).reduce((envVarsFromOptions, itm) => {
       const splitItm = itm.split('=');
       envVarsFromOptions[splitItm[0]] = splitItm.slice(1, splitItm.length).join('=') || '';
-    });
-    return envVarsFromOptions;
+      return envVarsFromOptions;
+    }, {});
   }
 
-  ensurePackage() {
+  async ensurePackage() {
     if (this.options['skip-package']) {
-      return fse
-        .access(
+      try {
+        await fse.access(
           path.join(this.serverless.config.servicePath, '.serverless', 'serverless-state.json')
-        )
-        .catch(() => this.serverless.pluginManager.spawn('package'));
+        );
+        return;
```

Comment:
`return` needs to be after `try/catch` (otherwise we're falling for `pluginManager.spawn` in case of exception)

Reply:
I think that the log is correct.

This is the original:
```js
    if (this.options['skip-package']) {
      return fse
        .access(
          path.join(this.serverless.config.servicePath, '.serverless', 'serverless-state.json')
        )
        .catch(() => this.serverless.pluginManager.spawn('package'));
    }
    return this.serverless.pluginManager.spawn('package');
```

if `fse.access` doesn't throw, `pluginManager.spawn` is never called;
if `fse.access` does throw, `pluginManager.spawn` is called in the catch.
The last `pluginManager.spawn` is called if `this.options['skip-package']` is false.

Maybe a simpler way to do the same would be to use the `fileExists` method which is one of your extension methods.

```js
    if (
      this.options['skip-package'] &&
      (await fileExists(
        path.join(this.serverless.config.servicePath, '.serverless', 'serverless-state.json')
      ))
    ) {
      return;
    }
    await this.serverless.pluginManager.spawn('package');
```

Wdyt?

Reply:
Indeed, I didn't notice that `this.serverless.pluginManager.spawn('package');` is doubled, so your construct perfectly reflects the original

> Maybe a simpler way to do the same would be to use the fileExists method which is one of your extension methods.

I will not go that route. Doing fs exist checks before doing an actual operation on file is coined as an anti-pattern (fact that file existed at fs.exists check, doesn't guarantee it'll exist on next call)